js/src/util.js

```diff
@@ -88,6 +88,22 @@ const Util = (() => {
           }
         }
       }
+    },
+
+    extend(obj1, obj2) {
```

Comment:
Why not use Object.assign?
I thing another issue is that this implementation is `deepCopy` but $.extend by default is not recursive.

Reply:
If Object.assign cannot be used I would use `arguments` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) and then you can change `Util.extend(Utils.extend...` to `Util.extend(obj1, obj2, obj3, ...)`

Reply:
I didn't had time to think more of this implementation of `extend` but it did the job 😄
I'll take a look thanks @wojtask9 👍 
If you want you can submit a PR which target this branch (`v4-carousel-without-jquery`) with a better implementation